ark/type/keywords/string/base64.ts

```diff
@@ -0,0 +1,20 @@
+import type { Predicate, of } from "../inference.ts"
+import { regexStringNode } from "./utils.ts"
+
+declare namespace string {
+	export type base64 = of<string, Predicate<"base64">>
+	export type base64url = of<string, Predicate<"base64url">>
```

Comment:
It seems like maybe `base64` itself should be a submodule like `string.json`, then `string.base64url` would be `string.base64.url`. Technically, the url wouldn't be a subtype of the root `string.base64` type, but already I've had to make some compromises on that to group things logically.

Would also allow easily adding a `string.base64.parse` in the future as suggested by Dimava if we arrive at a good output format for it.